gdown/download.py

```diff
@@ -232,13 +255,7 @@ def download(
             )
             raise FileURLRetrievalError(message)
 
-    if gdrive_file_id and is_gdrive_download_link:
-        content_disposition = urllib.parse.unquote(res.headers["Content-Disposition"])
-        m = re.search(r"filename\*=UTF-8''(.*)", content_disposition)
-        filename_from_url = m.groups()[0]
-        filename_from_url = filename_from_url.replace(osp.sep, "_")
-    else:
-        filename_from_url = osp.basename(url)
+    filename_from_url = get_filename_from_response(res) or osp.basename(url)
```

Comment:
Why you needed to change this part?

Reply:
Because the `Content-Disposition` headers are different in the response I received. If you look into the function I have implemented, the old behavior is still there and will be checked first. But if that doesn't work, as it doesn't for me, we fall back to a parsing strategy that works.

Reply:
I see. What about the check if it is a gdrive url?

Reply:
Good catch. Fixed in the latest commit.